### ISSUE TITLE
Where is the documentation?

### ISSUE THREAD

**Issue by thatJavaNerd** [ccrama/Slide#755]
As someone new to this project, I often find it hard to follow the code. There is no Javadoc. No wiki. Barely any comments. An open source project thrives on the community, but the harder it is for the community to get involved, the less they can do to help.

At the bare minimum, there should at least be Javadoc for the most important classes (`Reddit`, `Authentication`, etc.)

This was also brought up in #509 but it seems to have been ignored.

**Comment by okungnyo** [comment]
Maybe @ccrama is reluctant to let his baby leave the nest, if you know what I mean lol

**Comment by thatJavaNerd** [comment]
I hope for both @ccrama's and the project's sake that that's not the case.

**Comment by okungnyo** [comment]
Well, he does have a pro version and a donate option on the Play STore.

So I imagine the whole money issue would get kinda iffy on how that works if other people start to contribute more and he is no longer the main contributor.

But I am not a developer, and I have no idea how licenses work, so I am probably just talking out of my ass here.

**Comment by l3d00m** [comment]
Nah, I'm one of the most active contributors besides ccrama and I don't have a problem with money. I think I do a lot work for Slide, but it's still small compared to his. 

It's at least not an excuse why we shouldn't have a documentation.

**Comment by ccrama** [comment]
That is not at all the issue @okungnyo. Documentation is hard to do after the fact and that should have been something I did a long time ago. I'll be working on it more the coming weeks @thatJavaNerd , hopefully explain more what the classes actually do. 

As for the money comment, I am always open about what I make and would not restrict slide about money. Slide was never created to make me money (or it would be full of ads), and I have no issues reinvesting it into the community

**Comment by okungnyo** [comment]
Thanks for the clarification, and sorry for creating confusion.
